substrate/frame/revive/src/exec.rs

```diff
@@ -1451,12 +1448,18 @@ where
 				&self.first_frame.account_id,
 				contract.as_deref_mut(),
 			);
+
 			if let Some(contract) = contract {
 				AccountInfo::<T>::insert_contract(
 					&T::AddressMapper::to_address(&self.first_frame.account_id),
 					contract.clone(),
 				);
 			}
+			// destroy each contract registered for destruction
```

Comment:
This comment is redundant (it's obvious by looking at the next few lines of code).

I think what the comment should express instead is _why_ this needs to be done _exactly here_. This adds context and semantic information that is currently not already expressed explicitly by the code.